client/resource_group/controller/controller.go

```diff
@@ -280,6 +282,29 @@ func (c *ResourceGroupsController) Start(ctx context.Context) {
 
 		for {
 			select {
+			/* high priority */
```

Comment:
Moving the `case` statements to the front won't affect the priority. I think we should create a separate `select` for these high-priority cases.